apps/frontend/src/app/3_organisms/TransactionStepDialog/components/TransactionSteps/TransactionSteps.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
+
+import { ethers } from 'ethers';
+import { parseUnits } from 'ethers/lib/utils';
+
+import { Button, Icon, IconNames, StatusType } from '@sovryn/ui';
+
+import { APPROVAL_FUNCTION } from '../../../../../utils/constants';
+import { Transaction, TxConfig } from '../../TransactionStepDialog.types';
+import { TransactionStep } from '../TransactionStep/TransactionStep';
+
+export type TransactionStepsProps = {
+  transactions: Transaction[];
+  onSuccess?: () => void;
+  gasPrice: string;
+};
+
+export const TransactionSteps: FC<TransactionStepsProps> = ({
+  transactions,
+  onSuccess,
+  gasPrice,
+}) => {
+  const [configs, setConfigs] = useState<TxConfig[]>([]);
+  const [step, setStep] = useState(-1);
+  const [error, setError] = useState(false);
+
+  useEffect(() => {
+    const initilize = async () => {
+      const list: TxConfig[] = [];
+      for (let i = 0; i < transactions.length; i++) {
+        const tx = transactions[i];
+        const gasLimit = await tx.contract.estimateGas[tx.fnName](...tx.args);
+        list.push({
+          ...tx.config,
+          amount: tx.fnName === APPROVAL_FUNCTION ? tx.args[1] : undefined,
+          unlimitedAmount: tx.fnName === APPROVAL_FUNCTION ? false : undefined,
+          gasPrice,
+          gasLimit: gasLimit.toString(),
+        });
+      }
+      setConfigs(list);
+    };
+
+    if (gasPrice) {
+      initilize();
```

Comment:
```suggestion
    const initialize = async () => {
      const list: TxConfig[] = [];
      for (let i = 0; i < transactions.length; i++) {
        const tx = transactions[i];
        const gasLimit = await tx.contract.estimateGas[tx.fnName](...tx.args);
        list.push({
          ...tx.config,
          amount: tx.fnName === APPROVAL_FUNCTION ? tx.args[1] : undefined,
          unlimitedAmount: tx.fnName === APPROVAL_FUNCTION ? false : undefined,
          gasPrice,
          gasLimit: gasLimit.toString(),
        });
      }
      setConfigs(list);
    };

    if (gasPrice) {
      initialize();
```

Sorry, really small typo to update